src/components/templates/product-availability-modal/add-channels-modal-screen.tsx

```diff
@@ -0,0 +1,129 @@
+import { SalesChannel } from "@medusajs/medusa"
+import { useAdminSalesChannels } from "medusa-react"
+import React, { useState } from "react"
+import { usePagination, useRowSelect, useTable } from "react-table"
+import Button from "../../fundamentals/button"
+import Modal from "../../molecules/modal"
+import { LayeredModalContext } from "../../molecules/modal/layered-modal"
+
+import SalesChannelAvailabilityTable, {
+  useAvailableChannelsModalTableColumns,
+} from "./sales-channel-availability-table"
+
+export const useAddChannelsModalScreen = (
+  selectedRowIds,
+  onAddSelectedToAvailableChannels
+) => {
+  const { pop } = React.useContext(LayeredModalContext)
+
+  return {
+    title: "Add Sales Channels",
+    onBack: () => pop(),
+    view: (
+      <AddChannelsModalScreen
+        selectedSalesChannelIds={selectedRowIds}
+        onAddSelectedToAvailableChannels={onAddSelectedToAvailableChannels}
+      />
+    ),
+  }
+}
+
+type AddChannelsModalScreenProps = {
+  selectedSalesChannelIds: string[]
+  onAddSelectedToAvailableChannels: (salesChannels: SalesChannel[]) => void
+}
+
+const AddChannelsModalScreen: React.FC<AddChannelsModalScreenProps> = ({
+  onAddSelectedToAvailableChannels: onAddSelectedToAvailableChannels,
+  selectedSalesChannelIds,
+}) => {
+  const limit = 15
+  const [offset, setOffset] = useState(0)
+  const [query, setQuery] = useState("")
+  const [freeText, setFreeText] = useState("")
+
+  const { pop } = React.useContext(LayeredModalContext)
+  const [columns] = useAvailableChannelsModalTableColumns()
+
+  const { sales_channels: salesChannels } = useAdminSalesChannels({
+    q: freeText,
+  })
+
+  const numPages = Math.ceil((salesChannels?.length || 0) / limit)
+
+  const filterSalesChannels = (salesChannels: SalesChannel[]) => {
+    return salesChannels.filter((salesChannel) => {
+      return !selectedSalesChannelIds.includes(salesChannel.id)
+    })
+  }
+
+  const tableState = useTable(
```

Comment:
I lifted the tablestate because it was the same for both tables but how it was obtained changed (one using in memory channels, the other using the query).